config/config_test.go

```diff
@@ -177,6 +182,128 @@ func TestInitConfig(t *testing.T) {
+		err := os.WriteFile(continueFile1, []byte("TestVal: foo"), 0644)
+		require.NoError(t, err)
+		continueFile2 := filepath.Join(continueDir, "continue2.yaml")
+		err = os.WriteFile(continueFile2, []byte("TestVal: boo"), 0644)
+		require.NoError(t, err)
+
+		// Now we should pick up "boo" as the value, because continue2 is parsed last according to lexicographical order
+		err = handleContinuedCfg()
+		assert.NoError(t, err)
+		assert.Equal(t, "boo", viper.GetString("TestVal"))
+		assert.Equal(t, "bar", viper.GetString("otherVal"))
+	})
+
+	t.Run("test-two-continues-two-files", func(t *testing.T) {
+		viper.Reset()
+		continueDir1 := t.TempDir()
+		setupContConfig(t, continueDir1)
+
+		continueDir2 := t.TempDir()
+		continueFile1 := filepath.Join(continueDir1, "continue1.yaml")
+		// The first continued config will point to the second directory
+		err := os.WriteFile(continueFile1, []byte(fmt.Sprintf("ContinuedConfigLocation: %s\nTestVal: foo", continueDir2)), 0644)
+		require.NoError(t, err)
+
+		continueFile2 := filepath.Join(continueDir2, "continue2.yaml")
+		err = os.WriteFile(continueFile2, []byte("TestVal: boo"), 0644)
+		require.NoError(t, err)
+
+		// Start with dir 1, ensure we get to dir 2 and pick up the correct overwritten "boo" value
+		// We should pick up "boo" as the value, because continue2 is parsed last
+		err = handleContinuedCfg()
+		assert.NoError(t, err)
+		assert.Equal(t, "boo", viper.GetString("TestVal"))
+		assert.Equal(t, "bar", viper.GetString("otherVal"))
+	})
+
+	t.Run("test-circular-continue", func(t *testing.T) {
+		viper.Reset()
+		continueDir1 := t.TempDir()
+		setupContConfig(t, continueDir1)
+
+		continueDir2 := t.TempDir()
+		continueFile1 := filepath.Join(continueDir1, "continue1.yaml")
+		// The first continued config will point to the second directory
+		err := os.WriteFile(continueFile1, []byte(fmt.Sprintf("ContinuedConfigLocation: %s\nTestVal: foo", continueDir2)), 0644)
+		require.NoError(t, err)
+
+		continueFile2 := filepath.Join(continueDir2, "continue2.yaml")
+		err = os.WriteFile(continueFile2, []byte(fmt.Sprintf("ContinuedConfigLocation: %s\nTestVal: boo", continueDir1)), 0644)
+		require.NoError(t, err)
+
+		// We'll start with 1, go to 2, then re-check 1 to find we've already done it. The result is that
+		// we should pick up "boo" as the value, because continue2 is parsed last
+		err = handleContinuedCfg()
+		assert.NoError(t, err)
+		assert.Equal(t, "boo", viper.GetString("TestVal"))
+		assert.Equal(t, "bar", viper.GetString("otherVal"))
+	})
+
+	t.Run("test-bad-directory", func(t *testing.T) {
+		viper.Reset()
+		continueDir := t.TempDir()
+		setupContConfig(t, continueDir+"-dne")
+
+		continueFile := filepath.Join(continueDir, "continue.yaml")
+		err := os.WriteFile(continueFile, []byte("TestVal: foo"), 0644)
+		require.NoError(t, err)
+
+		err = handleContinuedCfg()
+		assert.Error(t, err)
+		assert.Contains(t, err.Error(), "does not exist")
+	})
```

Comment:
What if a user tries to continue a config dir that is empty or contains "config" files that are not `yaml`'s? I think we might want test cases for that.